### PR TITLE
Enum toJSON returns toString for a basic enum

### PR DESCRIPTION
**Pull request by amaurymartiny**
Before: for a basic Enum (non parametrized), toJSON returned toNumber, i.e. the index

After: it returns toString.

Why? Looks better, we can also move StorageHasher into interfaces. @jacogr i don't think this breaks anything right?

### CODE REVIEW

**Comment by jacogr** [comment]
It actually does break. I do agree it looks better.

The issue is (I believe, need to double check), that the JSON representation of a c-like enum is the index from the Rust side. The `toJSON` needs compatibility withy that.

**Comment by jacogr** [comment]
Wanted o post the folow-up yesterday and then didn' - been thinking abou a `toFormattted` that -

- formats enums nicely
- formats balances
- ... anything else that makes it human-readable

Never got around to it.

**Comment by amaurymartiny** [comment]
Yes, agreed, that would be a good idea. I'm closing this for now then.

File: packages/types/src/codec/Enum.spec.ts [hunk 1]
```diff
@@ -191,9 +191,9 @@ describe('Enum', (): void => {
     testDecode('Uint8Array', Uint8Array.from([0]), 'foo');
     testDecode('Uint8Array', Uint8Array.from([1]), 'bar');
 
-    testEncode('toJSON', 1);
+    testEncode('toJSON', 'bar');
     testEncode('toNumber', 1);
-    testEncode('toString', 'bar');
+    testEncode('toString', 'Bar');
```
Comment (amaurymartiny) [inline packages/types/src/codec/Enum.spec.ts]: Note: I didn't change the `.toString()`, that stays the same

**Comment by jacogr** [comment]
A conflict seems to have snuck in.

**Pull request merged**